src/it/scala/com.rbmhtechnology.calliope/TransactionalEventProducerSpec.scala

```diff
@@ -404,6 +405,22 @@ class TransactionalEventProducerSpec extends KafkaSpec with MustMatchers with Ty
             .expectNextN(2).map(e => e.value().payload) must contain inOrder(Event("1", "agg1"), Event("2", "agg2"))
         }
       }
+      "stop after being restarted" in {
+        val (topic, consumer) = topicConsumer()
+        val writer = runTransactionalEventProducer(topic, WritableEventStore(), settings)
+
+        whenReady(Future.sequence(Vector(
```

Comment:
You are right. Sending the message is not executed in a `Future` but it results in a `Future`, so we should be safe with the original impl. Sorry for the noise :-)